ohai/gpu_devices.rb

```diff
@@ -0,0 +1,24 @@
+Ohai.plugin(:GPUDevices) do
+  provides 'gpu_devices'
+
+  def gpu_devices_from_lspci
+    shell_out('lspci -vmm').stdout.split("\n\n").select do |pcidev|
+      pcidev =~ /(?:3D|VGA)(?: compatible)? controller/
+    end.map do |gpudev|
+      Mash.new.tap do |properties|
+        gpudev.lines.map do |line|
+          line.chomp!
+          key, val = line.split(":\t")
+          properties[key.downcase] = val
+        end
+      end
+    end
+  end
+
+  collect_data(:linux) do
+    gpu_devices Mash.new
+    gpu_devices_from_lspci.each do |dev|
+      gpu_devices[dev['slot']] = dev
+    end
```

Comment:
Ohai attributes are generally all object/record/hash/dict formatted rather than list/array formatted so that different data sources can uniquely identify each attribute. As a result, I'm storing the gpu_devices not as a list, but as a hash (using Ohai's Mash class which is just a fancy Hash) with the `slot` value as the key, since this is the only attribute guaranteed to be unique on a system (no two PCI devices can occupy the same slot, I hope) whereas it's possible to have multiple identical GPUs on a single host so I could not use the `device` field.